server/src/security/accessApiApprentissageService.ts

```diff
@@ -0,0 +1,31 @@
+import Boom from "boom"
+import jwt from "jsonwebtoken"
+import { z } from "zod"
+
+import config from "@/config"
+
+const jwtPublicKey = config.auth.apiApprentissage.publicKey
+
+const ZApiApprentissageTokenData = z.object({
+  email: z.string().email(),
+})
+
+export type ApiApprentissageTokenData = z.output<typeof ZApiApprentissageTokenData>
+
+export const parseApiApprentissageToken = (jwtToken: string): ApiApprentissageTokenData => {
+  try {
+    const { payload } = jwt.verify(jwtToken, jwtPublicKey, {
+      complete: true,
+      algorithms: ["ES512"],
+    })
+    const apiData = ZApiApprentissageTokenData.parse(payload)
+    return apiData
+  } catch (err: any) {
+    const errorStr = err + ""
+    if (errorStr === "TokenExpiredError: jwt expired") {
+      throw Boom.forbidden("JWT expired")
+    }
+    console.warn("invalid jwt token", jwtToken, err)
+    throw Boom.forbidden()
+  }
```

Comment:
Est-ce qu'on veut garder le `console.warn` ? Au pire on utilise le logger ?

C'est pas plus consistent de faire:

```suggestion
} catch (err: unknown) {
    // TokenExpiredError & JsonWebTokenError imported from jsonwebtoken
    if (err instanceof JsonWebTokenError || err instanceof TokenExpiredError) {
      const e = Boom.forbidden(err.message);
      e.cause = err;
      throw e;
    }
    
    // Ici peut etre throw un truc different en cas de zod errror 🤔 
    
    const e = Boom.internal();
    e.cause = err;
    throw e;
}
```

Reply:
En gros ici c'est surtout le fait de cast en string l'erreur qui me pose problème. Je conseil d'utiliser instanceof ou alors directement `err.name` https://www.npmjs.com/package/jsonwebtoken#tokenexpirederror

Aussi je conseil de mettre la cause toujours quand on throw ça permet à Sentry d'avoir la source meme si on s'en fou pour les forbidden (pas envoyé sur sentry).


Par contre le cas par defaut doit etre 500, c'est pas un cas expected. Par exemple une syntax erreur, ça permet de le retrouver dans Sentry.

Reply:
j'ai fait un refactoring